python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,380 @@
+                    g, feat, tmp=tmp, training=True
+                )
+
+                self.edge_mask = edge_mask
+
+                loss_tmp = self.loss(prob.unsqueeze(dim=0), ori_pred_dict[idx])
+                loss_tmp.backward()
+
+                loss += loss_tmp.item()
+                pred_label = prob.argmax(-1).item()
+                pred_list.append(pred_label)
+
+            optimizer.step()
+            print(f"Epoch: {epoch} | Loss: {loss}")
+
+    def explain_graph(self, graph, feat, tmp=1.0, training=False):
+        r"""Learn and return an an edge mask that play a crucial role to
+        explain the prediction made by the GNN for a graph. Also, return
+        the prediction made with just the edge mask.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            A homogeneous graph.
+        feat : Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        tmp : float
+            The temperature parameter fed to the sample procedure
+        training : bool
+            Indicates whether the concrete_sample is called during
+            training or evaluation.
+
+        Returns
+        -------
+        Tensor, Tensor
+            The classification probability for graph with edge mask,
+            the probability mask for graph edges.
+
+        Examples
+        --------
+
+        >>> import torch as th
+        >>> import torch.nn as nn
+        >>> import dgl
+        >>> from dgl.data import GINDataset
+        >>> from dgl.dataloading import GraphDataLoader
+        >>> from dgl.nn import GraphConv, PGExplainer
+
+        >>> # Define the model
+        >>> class Model(th.nn.Module):
+        ...     def __init__(self, in_feats, out_feats):
+        ...         super().__init__()
+        ...         self.conv = GraphConv(in_feats, out_feats)
+        ...         self.fc = nn.Linear(out_feats, 1)
+        ...         th.nn.init.xavier_uniform_(self.fc.weight)
+        ...
+        ...     def forward(self, g, h, embed=False, edge_weight=None):
+        ...         h = self.conv(g, h, edge_weight=edge_weight)
+        ...         if not embed:
+        ...             g.ndata['h'] = h
+        ...             hg = dgl.mean_nodes(g, 'h')
+        ...             return th.sigmoid(self.fc(hg))
+        ...         else:
+        ...             return h
+
+        >>> # Load dataset
+        >>> data = GINDataset('MUTAG', self_loop=True)
+        >>> dataloader = GraphDataLoader(data, batch_size=64, shuffle=True)
+
+        >>> # Train the model
+        >>> feat_size = data[0][0].ndata['attr'].shape[1]
+        >>> model = Model(feat_size, data.gclasses)
+        >>> criterion = nn.BCELoss()
+        >>> optimizer = th.optim.Adam(model.parameters(), lr=1e-2)
+        >>> for bg, labels in dataloader:
+        ...     logits = model(bg, bg.ndata['attr'])
```

Comment:
```suggestion
        ...     preds = model(bg, bg.ndata['attr'])
```
`logits` is used to describe the result before `sigmoid/softmax`, in which case you use `BCEWithLogitsLoss`.

Reply:
I see. Thank you for clearing this up.